src/legacy/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.ts

```diff
@@ -68,12 +68,13 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
     }
 
     const indexPattern = await getIndexPattern(savedObject);
+    const indexPatterns = indexPattern ? [indexPattern] : [];
```

Comment:
Consider using _.compact ?

Reply:
Do we really want this? We usually try not to use lodash in places where we can rather easy go with modern JS instead.

Reply:
@timroes I swapped back to using plain JS here because of TS problems with thinking _.compact could return undefined.